File: transfer-frontend/src/services/UsuarioService.js
import axios from "axios";

const USUARIOS_ENDPOINT = "http://localhost:8080/api/usuarios/"

class UsuarioService {    
    createUsuario(usuario){
        return axios.post(USUARIOS_ENDPOINT, usuario);
    }
    updateUsuario(usuarioId, usuario){
        return axios.put(USUARIOS_ENDPOINT + usuarioId, usuario);
    }
    deleteUsuario(usuarioId){
        return axios.delete(USUARIOS_ENDPOINT + usuarioId);
    }
    getUsuarios(){
        return axios.get(USUARIOS_ENDPOINT);
    }
    getUsuarioById(usuarioId){
        return axios.get(USUARIOS_ENDPOINT + usuarioId);
    }            
}

export default new UsuarioService()

File: transfer-backend/src/main/java/br/com/transfer/service/UsuarioService.java
package br.com.transfer.service;

import java.util.List;

import br.com.transfer.entity.Usuario;

public interface UsuarioService {
	
	Usuario createUsuario(Usuario usuario);

    Usuario updateUsuario(Integer usuarioId, Usuario usuario);

    void deleteUsuario(Integer usuarioId);

    List<Usuario> getUsuarios();

    Usuario getUsuarioBydId(Integer usuarioId);

}